### PR TITLE
Fix for test flake when sync waiting and an intermediate error occurs

### PR DESCRIPTION
**Pull request by rhuss**
This fix workaround the issue as described in https://github.com/knative/serving/issues/9727 but is useful in general to cancel an error window if a non-error condition event is received.

### CODE REVIEW

File: pkg/wait/wait_for_ready.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"io"
 	"time"
 
+	"github.com/apex/log"
```
Comment (mattmoor) [inline pkg/wait/wait_for_ready.go:22]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Review by maximilien — APPROVED** [review]
/ok-to-test
/retest

**Comment by dsimansk** [comment]
Thanks @rhuss for the thorough analysis and fix. IMO it looks good, but lets wait for other reviewers.

/approve

**Review by navidshaikh — APPROVED** [review]
/lgtm

**Pull request merged**